lib/ufe/UsdStageMap.cpp

```diff
@@ -31,32 +68,52 @@ UsdStageMap g_StageMap;
 
 void UsdStageMap::addItem(const Ufe::Path& path, UsdStageWeakPtr stage)
 {
-	fPathToStage[path] = stage;
-	fStageToPath[stage] = path;
+	// We expect a path to the proxy shape node, therefore a single segment.
+	auto nbSegments = path.nbSegments();
```

Comment:
One minor thing that I just noticed, It doesn't look like we are guarding UsdStageMap.cpp against UFE1.0.  

See https://github.com/Autodesk/maya-usd/blob/f53f29c7258bc1d99897d843dbfef2e0316dd702/lib/CMakeLists.txt#L214

Reply:
Yup, I think that's fine: UsdStageMap is required for proper UFE v1 use, and it doesn't depend on any UFE v2 features, so it should only be protected for UFE compilation, which it currently is, not against UFE v2.

Reply:
Makes sense. Thanks

Reply:
As discussed, completely missed your point, great catch!  Fixed in
https://github.com/Autodesk/maya-usd/pull/237/commits/e508bdb626a80dcf58d08e17c85867514a36dd5b